### PR TITLE
Timestamp displayed in the bucket event notifications received

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Automates bz: 1958818

Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py [hunk 1]
```diff
@@ -147,4 +151,25 @@ def test_rgw_kafka_notifications(self, bucket_factory):
...
+        event_time = event_obj.get("Records")[0].get("eventTime")
+        format_string = "%Y-%m-%dT%H:%M:%S.%fZ"
+        try:
+            datetime.strptime(event_time, format_string)
+        except ValueError as ef:
+            log.error(
+                f"Timestamp event {event_time} doesnt match the pattern {format_string}"
+            )
+            raise ef
+
```
Comment (PrasadDesala) [inline tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:174]: Please add a log info message if the timestamp event matches the expected pattern

**Review by parikshithb — APPROVED** [review]
lgtm

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py [hunk 2]
```diff
@@ -25,6 +28,7 @@
 
 @tier1
 @bugzilla("1937187")
+@bugzilla("1958818")
```
Comment (hnallurv) [inline tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:31]: What is the polarion test id for this? Could you please create one if not done yet?

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py [hunk 3]
```diff
@@ -147,4 +151,25 @@ def test_rgw_kafka_notifications(self, bucket_factory):
...
+            f"bin/kafka-console-consumer.sh --bootstrap-server {constants.KAFKA_ENDPOINT} "
+            f"--topic {self.kafka_topic.name} --from-beginning --timeout-ms 20000"
+        )
+        pod_list = get_pod_name_by_pattern(
+            pattern="my-cluster-zookeeper", namespace=constants.AMQ_NAMESPACE
+        )
+        zookeeper_obj = get_pod_obj(name=pod_list[0], namespace=constants.AMQ_NAMESPACE)
+        event_obj = zookeeper_obj.exec_cmd_on_pod(command=cmd)
+        log.info(f"Event obj: {event_obj}")
+        event_time = event_obj.get("Records")[0].get("eventTime")
```
Comment (hnallurv) [inline tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:165]: How will this work in an upgraded system - 4.6.4 to 4.6.5 or 4.7 with target version having the fix? will it get old format or new format?..

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:31]: will modify/add the steps to the existing polarion test case

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py (hunk 3 above)
Comment (Akarsha-rai) [reply tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:165]: Thanks, at least for now that validation needs to be added that ceph image contains a fix else test fails

File: tests/e2e/workloads/amq/test_rgw_kafka_notifications.py [hunk 4]
```diff
@@ -147,4 +151,27 @@ def test_rgw_kafka_notifications(self, bucket_factory):
...
```
Comment (PrasadDesala) [inline tests/e2e/workloads/amq/test_rgw_kafka_notifications.py:165]: Should we check for the specific timestamp pattern only from 4.8 onwards?

Comment (Akarsha-rai) [reply tests/e2e/workloads/amq/test_rgw_kafka_notifications.py]: Added condition

**Review by PrasadDesala — APPROVED** [review]

**Review by shreekarSS — APPROVED** [review]
lgtm

**Pull request merged**